### PR TITLE
[MAINTENANCE] Placing AWS RedShift Imports Under Compatibility Pattern

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Instead of having plain imports of `sqlalchemy_redshift`, they are placed under the established "Compatibility" coding pattern.

### Remarks
* JIRA: DX-1/DX-440

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/expectations/metrics/util.py [hunk 1]
```diff
@@ -258,12 +250,15 @@ def _get_dialect_type_module(dialect=None):
...
+
+    # Redshift does not (yet) export types to top level; only recognize base SA types
+    # noinspection PyUnresolvedReferences
+    if aws.redshiftdialect and isinstance(
+        dialect,
+        aws.redshiftdialect.RedshiftDialect,
+    ):
+        return dialect.sa
+    else:
         pass
```
Comment (TrangPham) [inline great_expectations/expectations/metrics/util.py]: nit (non-blocking): I don't think the `else pass` is required

```suggestion
```

File: great_expectations/compatibility/aws.py [hunk 2]
```diff
@@ -25,3 +25,13 @@
     from botocore import exceptions
 except ImportError:
     exceptions = AWS_NOT_IMPORTED
+
+try:
+    import sqlalchemy_redshift
+except ImportError:
+    sqlalchemy_redshift = AWS_NOT_IMPORTED
```
Comment (TrangPham) [inline great_expectations/compatibility/aws.py]: I think within the same file is fine. Perhaps a different error message would be better as to not tell user to install something they may not require
```suggestion
    sqlalchemy_redshift =  NotImported(
    "Redshift connection components are not installed, please 'pip install sqlalchemy_redshift'"
)
```

File: great_expectations/expectations/metrics/util.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/metrics/util.py]: @TrangPham Great catch -- deleted.  Thanks!

File: great_expectations/compatibility/aws.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/compatibility/aws.py]: @TrangPham Different message created -- thanks!

**Review by TrangPham — APPROVED** [review]

**Pull request merged**